Add cross/fuse, cross/sshfs and spk/sshfs

sea3pea0 opened the pull request:
Added fuse and sshfs-fuse to spksrc/cross and added sshfs-fuse to spksrc/spk

I tested this with a DS412+, I have a working sshfs mount.

Dr-Bean commented on cross/fuse/Makefile:
```diff
@@ -0,0 +1,17 @@
+PKG_VERS = 2.9.4
+PKG_EXT = tar.gz
+PKG_DIST_NAME = $(PKG_NAME)-$(PKG_VERS).$(PKG_EXT)
+PKG_DIST_SITE = http://sourceforge.net/projects/fuse/files/fuse-2.X/2.9.4
```
Replace `fuse` with $(PKG_NAME). Same with `2.9.4` and $(PKG_VERS)

Dr-Bean commented on cross/fuse/Makefile:
```diff
@@ -0,0 +1,17 @@
+
+DEPENDS =
+
+HOMEPAGE = http://fuse.sourceforge.net/sshfs.html
```
Wrong link for fuse, isn't it?

Dr-Bean commented on cross/fuse/Makefile:
```diff
@@ -0,0 +1,17 @@
+
+HOMEPAGE = http://fuse.sourceforge.net/sshfs.html
+COMMENT  = File System in User Space
+LICENSE  =
```
Add a license here please.

Dr-Bean commented on cross/fuse/PLIST:
```diff
@@ -0,0 +1,11 @@
+bin:bin/fusermount
+bin:bin/ulockmgr_server
+rsc:include/fuse/fuse.h
```
Drop header files (`.h`), unless needed.

Dr-Bean commented on cross/fuse/PLIST:
```diff
@@ -0,0 +1,11 @@
+lnk:lib/libulockmgr.so
+lnk:lib/libulockmgr.so.1
+lib:lib/libulockmgr.so.1.0.1
+rsc:lib/pkgconfig/fuse.pc
```
Can be removed.

Dr-Bean commented on cross/fuse/patch/002-somepatch.patch:
```diff
@@ -0,0 +1,24 @@
+--- util/Makefile.in.org    2015-08-19 10:56:51.834947550 +0200
```
Great looking patch :P The name could use some work though...how about `cross-compile.patch`...something a bit more descriptive.

Dr-Bean commented on cross/sshfs-fuse/Makefile:
```diff
@@ -0,0 +1,17 @@
+PKG_VERS = 2.5
+PKG_EXT = tar.gz
+PKG_DIST_NAME = $(PKG_NAME)-$(PKG_VERS).$(PKG_EXT)
+PKG_DIST_SITE = http://sourceforge.net/projects/fuse/files/sshfs-fuse/2.5
```
Replace `sshfs-fuse` with $(PKG_NAME). Same with `2.5` and $(PKG_VERS)

Dr-Bean commented on cross/sshfs-fuse/Makefile:
```diff
@@ -0,0 +1,17 @@
+PKG_DIST_SITE = http://sourceforge.net/projects/fuse/files/sshfs-fuse/2.5
+PKG_DIR = $(PKG_NAME)-$(PKG_VERS)
+
+DEPENDS = cross/fuse cross/glib cross/zlib cross/libffi
```
I don't think `cross/zlib` and `cross/libffi` belong here.

Dr-Bean commented on cross/sshfs-fuse/Makefile:
```diff
@@ -0,0 +1,17 @@
+
+HOMEPAGE = http://fuse.sourceforge.net/sshfs.html
+COMMENT  = File System in User Space
+LICENSE  =
```
Add a license please.

Dr-Bean commented on cross/sshfs-fuse/Makefile:
```diff
@@ -0,0 +1,17 @@
+LICENSE  =
+
+GNU_CONFIGURE = 1
+CONFIGURE_ARGS = 
```
Remove this entry if unused.

Dr-Bean commented on cross/sshfs-fuse/PLIST:
```diff
@@ -0,0 +1,83 @@
+bin:bin/gdbus
+bin:bin/gdbus-codegen
+bin:bin/gio-querymodules
+bin:bin/glib-compile-resources
```
This PLIST needs work ;) It currently contains stuff that is created by glib, libffi etc. 
Remove everything _unless created directly_ by `sshfs-fuse`. Usually, the sshfs-fuse.plist file is a good starting point, but it will probably show entries that are not already in other plists.

Dr-Bean commented on spk/sshfs-fuse/Makefile:
```diff
@@ -0,0 +1,24 @@
+SPK_REV = 1
+SPK_ICON = src/sshfs-fuse.png
+
+DEPENDS = cross/$(SPK_NAME) cross/fuse cross/glib cross/zlib cross/libffi
```
For next time, entries should be alphabetical, with `cross/$(SPK_NAME)` last.
However, in this case, you might realize that all the entries here are dependencies of the cross package `sshfs-fuse`, but are not dependencies of the SPK package itself. You only need `DEPENDS = cross/$(SPK_NAME)` here.

Dr-Bean commented on spk/sshfs-fuse/Makefile:
```diff
@@ -0,0 +1,24 @@
+DEPENDS = cross/$(SPK_NAME) cross/fuse cross/glib cross/zlib cross/libffi
+
+MAINTAINER = SynoCommunity
+DESCRIPTION = This is a filesystem client based on the SSH File Transfer Protocol
```
It looks better if you say it like this: `sshfs is a filesystem [..]`

Dr-Bean commented on spk/sshfs-fuse/Makefile:
```diff
@@ -0,0 +1,24 @@
+
+MAINTAINER = SynoCommunity
+DESCRIPTION = This is a filesystem client based on the SSH File Transfer Protocol
+DESCRIPTION_GER = 
```
Drop if unused.

Dr-Bean commented on spk/sshfs-fuse/Makefile:
```diff
@@ -0,0 +1,24 @@
+MAINTAINER = SynoCommunity
+DESCRIPTION = This is a filesystem client based on the SSH File Transfer Protocol
+DESCRIPTION_GER = 
+RELOAD_UI = no
```
Drop if package is not startable.

Dr-Bean commented on spk/sshfs-fuse/Makefile:
```diff
@@ -0,0 +1,24 @@
+CHANGELOG = "Initial release"
+
+HOMEPAGE   = http://fuse.sourceforge.net/sshfs.html
+LICENSE    = 
```
Add a license please.

Dr-Bean commented on spk/sshfs-fuse/src/dsm-control.sh:
```diff
@@ -0,0 +1,24 @@
+
+# Package
+PACKAGE="sshfs-fuse"
+DNAME="sshfs-fuse"
```
`sshfs`, same as DISPLAY_NAME

Dr-Bean commented on spk/sshfs-fuse/src/installer.sh:
```diff
@@ -0,0 +1,50 @@
+
+# Package
+PACKAGE="sshfs-fuse"
+DNAME="sshfs-fuse"
```
See previous comment.

Dr-Bean commented:
Looking good. Three things, apart from the comments above:
- Add digests for your cross packages. Use `make digests`;
- I'd suggest to replace the icon with a larger one, 256x256;
- While you're updating, please add a newline to all the files that show the missing line-ending icon here on Github.

sea3pea0 commented:
I refined the plist files, added the licenses, added a new icon, added line feeds to end of files, removed some white space, change the name of that slick patch file ;) added a beta tag after the last commit.. Just 'git tag beta'?

cytec commented:
@sea3pea0 Beta tag has to be set in spk/ Makefile... see: https://github.com/SynoCommunity/spksrc/blob/master/spk/monit/Makefile#L8

Dr-Bean commented on cross/fuse/digests:
```diff
@@ -0,0 +1,6 @@
+fuse-2.9.4.tar.gz SHA1 c8b25419f33624dc5240af6a5d26f2c04367ca71
+fuse-2.9.4.tar.gz SHA256 6be9c0bff6af8c677414935f31699ea5a7f8f5f791cfa5205be02ea186b97ce1
+fuse-2.9.4.tar.gz MD5 ecb712b5ffc6dffd54f4a405c9b372d8
+fuse-2.9.4.tar.gz SHA1 c8b25419f33624dc5240af6a5d26f2c04367ca71
```
Please remove the current digests file and recreate, or manually remove the double entries. I think you ran `make digests` twice.

Dr-Bean commented on cross/fuse/patch/cross-compile.patch:
```diff
@@ -0,0 +1,25 @@
+--- util/Makefile.in.org    2015-08-19 10:56:51.834947550 +0200
```
I missed this the first time around, but the dir must be called `patches`, not `patch`. Didn't you run into this during testing? If I'm reading spksrc.patch.mk correctly, the patch won't be applied otherwise.

Dr-Bean commented on cross/sshfs/Makefile:
```diff
@@ -0,0 +1,17 @@
+DEPENDS = cross/fuse cross/glib
+
+HOMEPAGE = http://fuse.sourceforge.net/sshfs.html
+COMMENT  = SSHFS is a filesystem client based on the SSH File Transfer Protocol.
```
From what I've seen from the documentation, I believe the name of the client is referred to as `sshfs` (generally all lower case)
`SSHFS` is used for the actual filesystem itself. Can you verify that?

Dr-Bean commented on cross/sshfs/Makefile:
```diff
@@ -0,0 +1,17 @@
+
+HOMEPAGE = http://fuse.sourceforge.net/sshfs.html
+COMMENT  = SSHFS is a filesystem client based on the SSH File Transfer Protocol.
+LICENSE  = GNU GENERAL PUBLIC LICENSE Version 2, June 1991
```
Nit: You could call it `GNU GPLv2` :)

Dr-Bean commented on cross/sshfs/digests:
```diff
@@ -0,0 +1,6 @@
+sshfs-fuse-2.5.tar.gz SHA1 a28ed6c04847611816452441b14ad9a0bb6df842
+sshfs-fuse-2.5.tar.gz SHA256 e9171452e5d0150b9c6a2158fd2e2dcefb5d5d03ba4d208949e00a3a46c6e63e
+sshfs-fuse-2.5.tar.gz MD5 17494910db8383a366b1301e5f5148a9
+sshfs-fuse-2.5.tar.gz SHA1 a28ed6c04847611816452441b14ad9a0bb6df842
```
Either recreate (`rm -rf digests && make digests`), or simply remove the last three entries.

Dr-Bean commented on spk/sshfs/Makefile:
```diff
@@ -0,0 +1,24 @@
+DEPENDS = cross/$(SPK_NAME)
+
+MAINTAINER = SynoCommunity
+DESCRIPTION = SSHFS is a filesystem client based on the SSH File Transfer Protocol
```
See my other comment on lower case usage. Whatever the name, it should be the same throughout the package (as in DISPLAY_NAME, DNAME; everything a user of the package can come across during normal usage)

Dr-Bean commented on spk/sshfs/Makefile:
```diff
@@ -0,0 +1,24 @@
+BETA = 1
+
+HOMEPAGE   = http://fuse.sourceforge.net/sshfs.html
+LICENSE    = GNU GENERAL PUBLIC LICENSE Version 2, June 1991
```
Same as other comment.

Dr-Bean commented:
If I can't come up with anything else, this should be good to go I think ;)

sea3pea0 commented:
I think I completed everything. Thanks again for the help. A couple of potential improvements to this spk could be a gui accessible within DSM where you could mount folders and set them to be auto mounted after boot, as well as the ability to remove mounts. This spk is adequate for my needs, but I'm sure the improvements I mentioned would make it more useful for those users who don't want to mess around on the command line.

Dr-Bean commented:
You're either building with root permissions, or you didn't clean the work dir before testing without the patch. Fuse is putting files in places where you don't want them. Please include the patch in the /patches dir :)

sea3pea0 commented:
I don't know what happened, I guess I had the folder named patch so the patch wasn't working but yes I had root permissions :) Now that I tried building with non root it doesn't work. When I have the patch in the patches folder this is the [output](https://gist.github.com/sea3pea0/804f3e40b44054c71ea5) I get.
This is what is in [Makefile.in.rej](https://gist.github.com/sea3pea0/1cd5cc4f0d428266dc7e)

Dr-Bean commented:
Looks like Github changed the patch's line endings when I copy/pasted it in the sshfs issue, and it doesn't look I can correct that. Just recreate it.

sea3pea0 commented:
I'm sorry, how exactly should I recreate that file?

Dr-Bean commented:
Remove the current patchfile, compile until it gets stuck, then redo the changes. It looks like this:
```
/spksrc/cross/fuse$ mkdir -p patches ##Make sure the patches dir is empty!
/spksrc/cross/fuse$ make clean ##Clean up old stuff
rm -fr work work-*
/spksrc/cross/fuse$ make arch-88f6281-5.2 ##We need fuse's source to be extracted, this is an easy way.
===>  Building package for arch 88f6281
[..]
../../mk/spksrc.cross-cc.mk:122: recipe for target 'arch-88f6281' failed
make: [arch-88f6281] Error 2 (ignored) ##Yep, failed ;)
/spksrc/cross/fuse$ cd work-88f6281-5.2/fuse-2.9.4/ ##Enter $(WORK_DIR)/$(PKG_DIR), which is where the patch is applied from
/spksrc/cross/fuse/work-88f6281-5.2/fuse-2.9.4$ cp util/Makefile.in util/Makefile.in.org ##Create original file to compare with
/spksrc/cross/fuse/work-88f6281-5.2/fuse-2.9.4$ vi util/Makefile.in ##Open the file in your favorite editor
##..and make the necessary changes, meaning: redo manually what the broken patch does.
/spksrc/cross/fuse/work-88f6281-5.2/fuse-2.9.4$ diff -Naur util/Makefile.in.org util/Makefile.in > ../../patches/cross-compile.patch ##This creates the actual patch.
/spksrc/cross/fuse/work-88f6281-5.2/fuse-2.9.4$ cd ../.. ##And we're done here...
/spksrc/cross/fuse$ make clean ##...so clean up again...
rm -fr work work-*
/spksrc/cross/fuse$ make arch-88f6281-5.2 ##...and test.
===>  Building package for arch 88f6281
[..]
make[1]: Leaving directory '/spksrc/cross/fuse'
/spksrc/cross/fuse$ #Done
```
